src/layout/GlobalLayout/Locale.tsx

```diff
@@ -1,40 +1,61 @@
 import { ConfigProvider } from 'antd';
-import { PropsWithChildren, memo, useState } from 'react';
+import { PropsWithChildren, memo, useEffect, useRef, useState } from 'react';
 import useSWR from 'swr';
 
+import { DEFAULT_LANG } from '@/const/locale';
 import { createI18nNext } from '@/locales/create';
 import { useOnFinishHydrationGlobal } from '@/store/global';
 import { isOnServerSide } from '@/utils/env';
 import { switchLang } from '@/utils/switchLang';
 
 interface LocaleLayoutProps extends PropsWithChildren {
-  lang?: string;
+  defaultLang?: string;
 }
 
-const Locale = memo<LocaleLayoutProps>(({ children, lang }) => {
+const FETCH_ANTD_LOCALE = 'fetch-antd-locale';
+
+const Locale = memo<LocaleLayoutProps>(({ children, defaultLang }) => {
+  const [lang, setLang] = useState(defaultLang ?? DEFAULT_LANG);
+  const [i18n] = useState(createI18nNext(defaultLang));
+
   const { data: locale } = useSWR(
-    lang,
-    async () =>
-      await import(`antd/locale/${lang?.includes('-') ? lang?.replace('-', '_') : 'en_US'}.js`),
+    [FETCH_ANTD_LOCALE, lang],
+    async ([, l]) => {
+      const localeName = l?.includes('-') ? l?.replace('-', '_') : 'en_US';
+      const antdLocale = await import(`antd/locale/${localeName}.js`);
```

Comment:
i think you can use normalizeLocale function just import it  from locales/resources 
```
const antdLocale = await import(`antd/locale/${normalizeLocale(lang!).replace('-', '_')}.js`);
```
or 
```
 const localeName =normalizeLocale(l!).replace('-', '_');
 const antdLocale = await import(`antd/locale/${localeName}.js`);
 ```